SU2_CFD/src/iteration_structure.cpp

```diff
@@ -139,80 +139,80 @@ void CIteration::SetGrid_Movement(CGeometry ****geometry_container,
-      grid_movement[val_iZone][val_iInst]->SetVolume_Deformation(geometry_container[val_iZone][val_iInst][MESH_0],
-                                           config_container[val_iZone], true);
+      grid_movement->SetVolume_Deformation(geometry[MESH_0],
+                                           config, true);
 
       /*--- Plunging ---*/
 
       /*--- Compute the new node locations for moving markers ---*/
 
-      surface_movement[val_iZone]->Surface_Plunging(geometry_container[val_iZone][val_iInst][MESH_0],
-                                         config_container[val_iZone], ExtIter, val_iZone);
+      surface_movement->Surface_Plunging(geometry[MESH_0],
+                                         config, ExtIter, val_iZone);
       /*--- Deform the volume grid around the new boundary locations ---*/
 
       if (rank == MASTER_NODE)
         cout << " Deforming the volume grid." << endl;
-      grid_movement[val_iZone][val_iInst]->SetVolume_Deformation(geometry_container[val_iZone][val_iInst][MESH_0],
-                                           config_container[val_iZone], true);
+      grid_movement->SetVolume_Deformation(geometry[MESH_0],
+                                           config, true);
 
       /*--- Pitching ---*/
 
       /*--- Compute the new node locations for moving markers ---*/
 
-      surface_movement[val_iZone]->Surface_Pitching(geometry_container[val_iZone][val_iInst][MESH_0],
-                                         config_container[val_iZone], ExtIter, val_iZone);
+      surface_movement->Surface_Pitching(geometry[MESH_0],
+                                         config, ExtIter, val_iZone);
       /*--- Deform the volume grid around the new boundary locations ---*/
 
       if (rank == MASTER_NODE)
         cout << " Deforming the volume grid." << endl;
-      grid_movement[val_iZone][val_iInst]->SetVolume_Deformation(geometry_container[val_iZone][val_iInst][MESH_0],
-                                           config_container[val_iZone], true);
+      grid_movement->SetVolume_Deformation(geometry[MESH_0],
+                                           config, true);
 
       /*--- Rotating ---*/
 
       /*--- Compute the new node locations for moving markers ---*/
 
-      surface_movement[val_iZone]->Surface_Rotating(geometry_container[val_iZone][val_iInst][MESH_0],
-                                         config_container[val_iZone], ExtIter, val_iZone);
+      surface_movement->Surface_Rotating(geometry[MESH_0],
+                                         config, ExtIter, val_iZone);
       /*--- Deform the volume grid around the new boundary locations ---*/
 
       if (rank == MASTER_NODE)
         cout << " Deforming the volume grid." << endl;
-      grid_movement[val_iZone][val_iInst]->SetVolume_Deformation(geometry_container[val_iZone][val_iInst][MESH_0],
-                                           config_container[val_iZone], true);
+      grid_movement->SetVolume_Deformation(geometry[MESH_0],
```

Comment:
the previous 3 calls to SetVolume_Deformation() are the ones in question in #730 (this last one of the four should stay), can delete now if you like

Reply:
You can go ahead and do that if you want. Do you think it works by just removing them ?

Reply:
You're right - let's not introduce any new issues here by accident. I can look at the grid movement change once all of the refactoring is in place.